### PR TITLE
Adds XML exporter

### PR DESCRIPTION
**Pull request by patriksvensson**
This PR adds support for exporting data as XML. I more or less looked at how the console visualizer does things and replicated that behavior for the XML export.

This is done after my own needs so I'm of course open to change anything that you don't think aligns with AsmSpy in any way.

### CODE REVIEW

File: AsmSpy.CommandLine/XmlExport.cs [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+            _result = result ?? throw new ArgumentNullException(nameof(result));
+            _fileName = fileName?.Trim();
+            _logger = logger ?? throw new ArgumentNullException(nameof(logger));
+        }
+
+        public void Visualize()
+        {
+            if (string.IsNullOrWhiteSpace(_fileName))
+            {
+                _logger.LogError("No valid filename specified.");
```
Comment (patriksvensson) [inline AsmSpy.CommandLine/XmlExport.cs]: Return here

**Comment by mikehadlow** [comment]
Thanks @patriksvensson, looks good. Can you also update the readme.md file so that the new feature is documented. Cheers!

**Comment by patriksvensson** [comment]
@mikehadlow Absolutely

**Comment by patriksvensson** [comment]
@mikehadlow I've changed the option name to `-x|--xml` to be more consistent with the DGML exporter. Also updated readme.

**Comment by mikehadlow** [comment]
Merged, thanks for this excellent addition @patriksvensson! Great work.

**Comment by patriksvensson** [comment]
@mikehadlow It was all my pleasure. Thanks for creating a really useful tool!

**Pull request merged**